### PR TITLE
Hotfix/document loader

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/commands/open.js [hunk 1]
```diff
@@ -19,9 +19,13 @@ export class Open extends Base {
 			elementor.$previewContents.find( `.elementor-${ id }` ).addClass( 'loading' );
 		}
 
+		const startAutoSave = this.component.startAutoSave;
+
```
Comment (iNewLegend) [inline assets/dev/js/editor/commands/open.js]: Remove

**Review by iNewLegend — COMMENTED** [review]
Change

File: assets/dev/js/editor/document/save/component.js [hunk 2]
```diff
@@ -1,15 +1,13 @@
 import BackwardsCompatibility from './backwards-compatibility';
-import * as Commands from './commands';
+import * as commands from './commands/';
 
 export default class Component extends BackwardsCompatibility {
 	__construct( args = {} ) {
 		super.__construct( args );
 
-		this.autoSaveTimer = null;
```
Comment (matipojo) [inline assets/dev/js/editor/document/save/component.js:8]: BC

File: assets/dev/js/editor/document/save/component.js [hunk 3]
```diff
@@ -71,25 +69,13 @@ export default class Component extends BackwardsCompatibility {
 		this.trigger( 'save', options );
 	}
 
-	startTimer( hasChanges ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/save/component.js:74]: BC

File: assets/dev/js/editor/editor-base.js [hunk 4]
```diff
@@ -738,10 +737,6 @@ export default class EditorBase extends Marionette.Application {
 
 		this.initComponents();
 
-		this.documents = $e.components.register( new EditorsDocument() );
-
-		this.saver = $e.components.get( 'document/save' );
```
Comment (matipojo) [inline assets/dev/js/editor/editor-base.js:746]: BC

File: core/common/assets/js/api/modules/component-base.js [hunk 5]
```diff
@@ -13,6 +13,8 @@ export default class ComponentBase extends elementorModules.Module {
 		this.currentTab = '';
 	}
 
+	onInit() {}
```
Comment (matipojo) [inline core/common/assets/js/api/modules/component-base.js]: Already exist? see `elementorModules.Module`

File: tests/qunit/core/common/assets/js/api/core/components.spec.js [hunk 6]
```diff
@@ -46,17 +46,17 @@ jQuery( () => {
 		QUnit.test( 'Register routes via tabs', ( assert ) => {
 			const namespace = 'register-via-tabs',
 				Component = class extends ComponentBase {
-					getNamespace() {
-						return namespace;
-					}
+				getNamespace() {
```
Comment (matipojo) [inline tests/qunit/core/common/assets/js/api/core/components.spec.js:48]: Why?

File: tests/qunit/core/common/assets/js/api/core/components.spec.js [hunk 7]
```diff
@@ -46,17 +46,17 @@ jQuery( () => {
...
+				}
 
-					defaultTabs() {
-						return {
-							tabA: { title: 'tabA' },
-							tabB: { title: 'tabB' },
-						};
-					}
-				};
+				defaultTabs() {
```
Comment (matipojo) [inline tests/qunit/core/common/assets/js/api/core/components.spec.js:52]: Why?

File: assets/dev/js/editor/editor-document.js [hunk 8]
```diff
@@ -15,4 +15,9 @@ elementorCommon.elements.$window.on( 'elementor:init-components', () => {
 	$e.components.register( new RepeaterComponent() );
 	$e.components.register( new HistoryComponent() );
 	$e.components.register( new DynamicComponent() );
+
+	elementor.documents = $e.components.register( new EditorDocuments() );
+
+	// TODO: Remove, BC Since 2.9.0.
+	elementor.saver = $e.components.get( 'document/save' );
```
Comment (matipojo) [inline assets/dev/js/editor/editor-document.js]: `defineProperty` as deprecated.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**